python-backend/app/api/mines/variances/resources/variance.py

```diff
@@ -0,0 +1,135 @@
+from flask_restplus import Resource, reqparse, fields
+from sqlalchemy.exc import DBAPIError
+from werkzeug.exceptions import BadRequest
+
+from ..models.variance import Variance
+from app.extensions import api
+from ....utils.access_decorators import requires_role_mine_view, requires_role_mine_create
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+
+mine_document_model = api.model('MineDocument', {
+    'mine_document_guid': fields.String,
+    'mine_guid': fields.String,
+    'document_manager_guid': fields.String,
+    'document_name': fields.String,
+    'active_ind': fields.Boolean
+})
+
+variance_document_model = api.model('VarianceDocument', {
+    'variance_document_xref_guid': fields.String,
+    'variance_id': fields.Integer,
+    'mine_document_guid': fields.String,
+    'details': fields.Nested(mine_document_model)
+})
+
+variance_model = api.model('Variance', {
+    'variance_id': fields.Integer,
+    'compliance_article_id': fields.Integer,
+    'note': fields.String,
+    'issue_date': fields.Date,
+    'received_date': fields.Date,
+    'expiry_date': fields.Date,
+    'related_documents': fields.List(fields.Nested(variance_document_model))
+})
+
+class VarianceListResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('compliance_article_id',
+                        type=int,
+                        store_missing=False,
+                        trim=True,
+                        help='ID representing the MA or HSRCM code to which this variance relates.')
+    parser.add_argument('note',
+                        type=str,
+                        store_missing=False,
+                        trim=True,
+                        help='A note to include on the variance. Limited to 300 characters.')
+    parser.add_argument('issue_date',
+                        store_missing=False,
+                        trim=True,
+                        help='The date on which the variance was issued.')
+    parser.add_argument('received_date',
+                        store_missing=False,
+                        trim=True,
+                        help='The date on which the variance was received.')
+    parser.add_argument('expiry_date',
+                        store_missing=False,
+                        trim=True,
+                        help='The date on which the variance expires.')
+
+
+    @api.doc(
+        description=
+        'Get a list of all variances for a given mine.',
+        params={
+            'mine_guid': 'guid of the mine for which to fetch variances'
+        }
+    )
+    @requires_role_mine_view
+    @api.marshal_with(variance_model, code=200, envelope='records')
+    def get(self, mine_guid):
+        try:
+            variances = Variance.find_by_mine_guid(mine_guid)
+        except DBAPIError:
+            raise BadRequest('Invalid mine_guid')
+
+        if variances is None:
+            raise BadRequest('Unable to fetch variances')
```

Comment:
Do we want to throw an error, seems like we should just return an empty list?

Reply:
It'll be an empty list if there were no variances on the mine. If the query returns `None` then something went wrong with the query, which is why I figured 'Unable to fetch variances' made the most sense.

Not married to this perspective, but it seems logical to me.

Reply:
oh right, you didn't check falsy, you check for None... yea that check doesn't hurt, disregard.